brickflow/engine/workflow.py

```diff
@@ -327,6 +327,30 @@ def _add_task(
         else:
             ensure_plugins = ensure_brickflow_plugins
 
+        # NOTE: REMOTE WORKSPACE RUN JOB OVERRIDE
+        # This is a temporary override for the RunJobTask because Databricks does not natively support
+        # triggering the job run in the remote workspace. By default, Databricks SDK derives the workspace URL
+        # from the runtime, and hence it is not required by the RunJobTask. The assumption is that if `host` parameter
+        # is set, user wants to trigger a remote job, in this case we set the task type to BRICKFLOW_TASK to
+        # enforce notebook type execution and replacing the original callable function with the RunJobInRemoteWorkspace
+        if task_type == TaskType.RUN_JOB_TASK:
+            func = f()
+            if func.host:
+                from brickflow_plugins.databricks.run_job import RunJobInRemoteWorkspace
```

Comment:
Though I didn't want plugins to be used in the core module, this makes sense as it's temporary and in future the team don't need to change their code as much when databricks natively provides this support!